integration-tests/src/test/java/tachyon/master/MasterClientAuthenticationIntegrationTest.java

```diff
@@ -0,0 +1,201 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.master;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import javax.security.sasl.AuthenticationException;
+import javax.security.sasl.SaslException;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import tachyon.Constants;
+import tachyon.conf.TachyonConf;
+import tachyon.security.LoginUser;
+import tachyon.security.authentication.AuthenticationProvider;
+import tachyon.security.authentication.AuthenticationFactory.AuthType;
+
+/**
+ * Though its name indicates that it provides the tests for Tachyon authentication.
+ * This class is likely to test four authentication modes: NOSASL, SIMPLE, CUSTOM, KERBEROS.
+ *
+ */
+public class MasterClientAuthenticationIntegrationTest {
+  private LocalTachyonCluster mLocalTachyonCluster = null;
+  private ExecutorService mExecutorService = null;
+  private TachyonConf mMasterTachyonConf = null;
+  private MasterInfo mMasterInfo = null;
+
+  @Rule
+  public ExpectedException mThrown = ExpectedException.none();
+
+  @Before
+  public void before() throws Exception {
+    mLocalTachyonCluster = new LocalTachyonCluster(1000, 1000, Constants.GB);
+    mMasterTachyonConf = mLocalTachyonCluster.getMasterTachyonConf();
+    mExecutorService = Executors.newFixedThreadPool(2);
+  }
+
+  @Test
+  public void noAuthenticationOpenCloseTest() throws Exception {
+    //no authentication configure
+    setAuthenticationType(AuthType.NOSASL);
+    //start cluster
+    mLocalTachyonCluster.start();
+
+    mMasterInfo = mLocalTachyonCluster.getMasterInfo();
+    MasterClient masterClient =
+        new MasterClient(mMasterInfo.getMasterAddress(),
+            mExecutorService, mMasterTachyonConf);
+    Assert.assertFalse(masterClient.isConnected());
+    masterClient.connect();
+    Assert.assertTrue(masterClient.isConnected());
+    masterClient.user_createFile("/file-nosasl", "", Constants.DEFAULT_BLOCK_SIZE_BYTE, true);
+    Assert.assertTrue(masterClient.getFileStatus(-1, "/file-nosasl") != null);
+    masterClient.disconnect();
+    Assert.assertFalse(masterClient.isConnected());
+    masterClient.connect();
+    Assert.assertTrue(masterClient.isConnected());
+    Assert.assertTrue(masterClient.getFileStatus(-1, "/file-nosasl") != null);
+    masterClient.close();
+
+    //stop cluster
+    mLocalTachyonCluster.stop();
+  }
+
+  @Test
+  public void simpleAuthenticationOpenCloseTest() throws Exception {
+    //simple authentication configure
+    setAuthenticationType(AuthType.SIMPLE);
```

Comment:
It seems `simpleAuthenticationOpenCloseTest()` has exactly the same test and expected results as `noAuthenticationOpenCloseTest`. I thought simple authentication allows any combination of username/password, shall we test that (through a few combinations)?